pkg/kv/kvserver/logstore/logstore.go

```diff
@@ -543,6 +545,16 @@ func LoadEntries(
 
 	ents, cachedSize, hitIndex, exceededMaxBytes := eCache.Scan(ents, rangeID, lo, hi, maxBytes)
 
+	budgetFor := func(size uint64) bool {
+		return account == nil || account.Grow(ctx, int64(size)) == nil
```

Comment:
Figure out what to do if `size` > limit. Probably should constrain the limit to be >= max command size if we have such.

Reply:
If `size > limit`, I think we probably want to make sure the budget has the entire `limit` available, and then allow exceeding it.

Does the byte monitor allow overuse?

Reply:
I don't think `BytesMonitor` allows overuse. We could do one of:
- extend `BytesMonitor` with a soft limit that can be exceeded once
- artificially lower the `size` to `limit` if it exceeds the `limit`, and accept that the metrics would be inaccurate.

Reply:
Extended the `BytesMonitor`, see the first commit.